ethereum/core/src/main/java/org/hyperledger/besu/ethereum/core/DepositContract.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright contributors to Hyperledger Besu
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.hyperledger.besu.ethereum.core;
+
+import org.hyperledger.besu.evm.log.LogTopic;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+import org.web3j.abi.TypeReference;
+import org.web3j.abi.datatypes.DynamicBytes;
+import org.web3j.abi.datatypes.Event;
+import org.web3j.protocol.Web3j;
+import org.web3j.protocol.core.methods.response.Log;
+import org.web3j.tx.Contract;
+import org.web3j.tx.TransactionManager;
+import org.web3j.tx.gas.ContractGasProvider;
+
+public class DepositContract extends Contract {
+
+  public static final Event DEPOSITEVENT_EVENT =
+      new Event(
+          "DepositEvent",
+          Arrays.asList(
+              new TypeReference<DynamicBytes>() {},
+              new TypeReference<DynamicBytes>() {},
+              new TypeReference<DynamicBytes>() {},
+              new TypeReference<DynamicBytes>() {},
+              new TypeReference<DynamicBytes>() {}));
+
+  protected DepositContract(
```

Comment:
Interestingly if I set it to `private`, it will complain the constructor is never used. I will leave it as `protected` unless there is a better way to make it compile

Reply:
package-private (i.e. removing `protected` keyword will work).
By "suppress warning" I meant the "never used" warning you're talking about. Protected is ok, package-private slightly better (in terms of minimising visibility)